server/util/consistent_hash/consistent_hash.go

```diff
@@ -38,21 +42,26 @@ func (c *ConsistentHash) GetItems() []string {
 	return c.items
 }
 
-func (c *ConsistentHash) Set(items ...string) {
+func (c *ConsistentHash) Set(items ...string) error {
+	if len(items) > 256 {
+		return status.InvalidArgumentError("Too many items in consistent hash, max allowed: 256")
+	}
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	c.keys = make([]int, 0)
-	c.ring = make(map[int]string, 0)
+	c.ring = make(map[int]uint8, 0)
```

Comment:
We should reset `replicaSets` because the existing cache data won't be useful if the items change.

Reply:
Done.